src/edges.c

```diff
@@ -182,27 +222,161 @@ validate_output_edges(struct border *valid_edges,
 	/* Left edge encounters a half-infinite region to the left of the output */
 
 	validate_single_output_edge(&valid_edges->left,
-			view, target, output, validator, VIEW_EDGE_LEFT);
+			view, target, output, validator, WLR_EDGE_LEFT);
 
 	/* Right edge encounters a half-infinite region to the right of the output */
 
 	validate_single_output_edge(&valid_edges->right,
-			view, target, output, validator, VIEW_EDGE_RIGHT);
+			view, target, output, validator, WLR_EDGE_RIGHT);
 
 	/* Top edge encounters a half-infinite region above the output */
 
 	validate_single_output_edge(&valid_edges->top,
-			view, target, output, validator, VIEW_EDGE_UP);
+			view, target, output, validator, WLR_EDGE_TOP);
 
 	/* Bottom edge encounters a half-infinite region below the output */
 	validate_single_output_edge(&valid_edges->bottom,
-			view, target, output, validator, VIEW_EDGE_DOWN);
+			view, target, output, validator, WLR_EDGE_BOTTOM);
+}
+
+/* Test if parts of the current view is covered by the remaining space in the region */
+static void
+subtract_view_from_space(struct view *view, pixman_region32_t *available)
+{
+	struct wlr_box view_size = ssd_max_extents(view);
+	pixman_box32_t view_rect = {
+		.x1 = view_size.x,
+		.x2 = view_size.x + view_size.width,
+		.y1 = view_size.y,
+		.y2 = view_size.y + view_size.height
+	};
+
+	pixman_region_overlap_t overlap =
+		pixman_region32_contains_rectangle(available, &view_rect);
+
+	switch (overlap) {
+	case PIXMAN_REGION_IN:
+		view->edges_visible = WLR_EDGE_TOP | WLR_EDGE_RIGHT
+			| WLR_EDGE_BOTTOM | WLR_EDGE_LEFT;
+		break;
+	case PIXMAN_REGION_OUT:
+		view->edges_visible = 0;
+		return;
+	case PIXMAN_REGION_PART:
+		{
```

Comment:
In other parts of the code we do also have
`; /* works around "a label can only be part of a statement" */`

as first line after the `case`, personally I'd prefer that variant in this case to get less indentation.

Reply:
Done